### ISSUE TITLE
explicit synchronization on concurrent collections

### ISSUE THREAD

**Issue by simonjenga** [yegor256/takes#470]
Concurrent collections are inherently thread-safe; they may be passed between threads or published without the need for explicit synchronization.
```
private final transient List<Response> transformed = new CopyOnWriteArrayList<Response>();

synchronized (this.transformed) {
	// to do
}
```
Please provide extensive reasons/explanations/documentation why this is being done on the following classes:
```
org.takes.rs.RsGzip;
org.takes.rs.RsPrettyJSON;
org.takes.rs.RsPrettyXML;

```

Otherwise these ```synchronized``` blocks look redundant and may be deleted while doing maintenance of the project since most of us don't know why they are there.

http://beginnersbook.com/2013/12/how-to-synchronize-arraylist-in-java-with-example/

**Comment by davvd** [comment]
@yegor256 dispatch this issue please, see [par.21](http://at.teamed.io/policy.html#21)

**Comment by yegor256** [comment]
@simonjenga this synchronization is not protecting the array, but its concurrent modification from multiple threads:

```
if (this.zipped.isEmpty()) {
  // boom! another thread adds something to this.zipped
  this.zipped.add(
  );
}
return this.zipped.get(0);
```

See?

**Comment by simonjenga** [comment]
@yegor256 I don't quite agree with you on this one. I also don't understand your explanation.

Your code is synchronizing on a concurrent collection!

What you need to protect from concurrent modification by multiple threads is the ```method```. It needs to be revised to be as follows:
```
private synchronized Response make() throws IOException {
	if (this.transformed.isEmpty()) {
		this.transformed.add(
			// to do
		);
	}        
	return this.transformed.get(0);
}
```

**Comment by yegor256** [comment]
@simonjenga using `synchronized` for a method is against PMD rule. See `AvoidSynchronizedAtMethodLevel` at http://pmd.sourceforge.net/pmd-4.3.0/rules/design.html

**Comment by simonjenga** [comment]
@yegor256 
````
synchronized void myMethod() {
    //code
}
```

is equivalent to:

```
void myMethod() {
    synchronized(this) {
        //code
    }
}
```
```

**Comment by yegor256** [comment]
@simonjenga not really. I quote PMD: "Method level synchronization can backfire when new code is added to the method. Block-level synchronization helps to ensure that only the code that needs synchronization gets it." We may eventually add more code to the method and that new code will also be synchronized. This is not what we want. Block level synchronization is more accurate.

**Comment by simonjenga** [comment]
@yegor256 OK, but there has to be some way to remove the explicit synchronizing on a concurrent collection!
